src/main/java/org/jabref/logic/net/URLDownload.java

```diff
@@ -384,7 +384,7 @@ public URLConnection openConnection() throws FetcherException {
             } else if (status >= 400) {
                 // in case of an error, propagate the error message
                 SimpleHttpResponse httpResponse = new SimpleHttpResponse(httpURLConnection);
-                LOGGER.info("{}", httpResponse);
+                LOGGER.info("{}: {}", this.source, httpResponse);
```

Comment:
This will leak api keys? I remember that we had an issue about this

Reply:
Quick fix at https://github.com/JabRef/jabref/pull/11852/commits/390d2ce7c86b3fa679b28aaf67b9756e5a961b0b